### PR TITLE
add averageAges solution

### PR DESCRIPTION
**Pull request by DidenkoDenis306**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -55,7 +67,22 @@ function calculateWomenAverageAge(people, withChildren) {
...
+    const mother = people.find(mom => mom.name === child.mother);
+
+    return child.born - mother.born;
+  });
+
+  return calculateAverageAge(ageDiff, children.length);
+}
+
+function calculateAverageAge(peopleAge, amountOfPeople) {
+  return peopleAge.reduce((acc, age) => (acc + age), 0) / amountOfPeople;
```
Comment (ruslanpashkov) [inline src/averageAges.js:85]: what if amount === 0?

File: src/averageAges.js [hunk 2]
```diff
@@ -15,11 +15,15 @@
...
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person =>
+    century
+      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+      : person.sex === 'm'
+  );
```
Comment (ruslanpashkov) [inline src/averageAges.js:22]:
```suggestion
  const men = people.filter(person => (
    person.sex === 'm' && (century
      ? Math.ceil(person.died / 100) === century
      : true)
  ));
```

File: src/averageAges.js [hunk 3]
```diff
@@ -55,7 +67,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    people.find(mother => mother.name === person.mother)
+    && (onlyWithSon ? person.sex === 'm' : true)
+  );
```
Comment (ruslanpashkov) [inline src/averageAges.js:73]:
```suggestion
  const children = people.filter(person => (
    people.find(mother => mother.name === person.mother) && (onlyWithSon
      ? person.sex === 'm'
      : true)
  ));
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/averageAges.js [hunk 4]
```diff
@@ -15,11 +15,15 @@
...
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => (
+    person.sex === 'm' && (century
+      ? Math.ceil(person.died / 100) === century
+      : true)
+  ));
+
+  const menAverageAge = men.map(man => man.died - man.born);
```
Comment (Anna-Tkachenko) [inline src/averageAges.js:24]: It's not `menAverageAge`, check variable name
Here is stored the set of mens` ages only

File: src/averageAges.js [hunk 5]
```diff
@@ -37,7 +41,15 @@ function calculateMenAverageAge(people, century) {
...
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => (
+    withChildren
+      ? people.find(child => child.mother === person.name)
+      : person.sex === 'f'
+  ));
+
+  const womenAverageAge = women.map(woman => woman.died - woman.born);
```
Comment (Anna-Tkachenko) [inline src/averageAges.js:50]: Same here, check variable name

File: src/averageAges.js [hunk 6]
```diff
@@ -55,7 +67,25 @@ function calculateWomenAverageAge(people, withChildren) {
...
+  const ageDiff = children.map(child => {
+    const mother = people.find(mom => mom.name === child.mother);
+
+    return child.born - mother.born;
+  });
+
+  return calculateAverageAge(ageDiff, children.length);
+}
+
+function calculateAverageAge(peopleAge, amountOfPeople) {
```
Comment (Anna-Tkachenko) [inline src/averageAges.js:85]: Will be better to use `number` for count nouns
```suggestion
function calculateAverageAge(peopleAge, numberOfPeople) {
```

Comment (Anna-Tkachenko) [reply src/averageAges.js:85]: Great job with moving repeated logic to separate function⭐️

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

**Review by 33Orange — APPROVED** [review]
Well done ! 👍 

All comments was fixed.

**Pull request closed without merging**